### PR TITLE
V2.0.0, refactored to add config options and table creation

### PR DESCRIPTION
**Pull request by bflorian**
Refactored to provide more flexible configuration options to better support single table
schemas and alternative ways of configuring AWS credentials. 

Also added support for a `state` table entry for storing arbitrary installed app data (though SDK changes will be needed to expose that to SmartApps)

This is a breaking change as far as configuring the context store is concerned, but compatible with the
older version and existing SDK as far as reading and writing context is concerned.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,31 +14,114 @@ The context stored by this module consists of the following data elements:
 * **refreshToken**: the refresh token used in generating a new access token when one expires
 * **clientId**: the SmartApp's client ID, used in generating a new access token
 * **clientSecret**: the SmartApp's client secret, used in generating a new access token
```
Comment (rossiam) [inline README.md]: These two fields aren't stored anymore, right?

File: lib/DynamoDBContextStore.js [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+				if (err) {
+					reject(err);
+				} else {
+					if (data && data.Item) {
+						const item = data.Item
+						resolve({
+							installedAppId: AWS.DynamoDB.Converter.output(item.installedAppId),
+							locationId: AWS.DynamoDB.Converter.output(item.locationId),
+							authToken: AWS.DynamoDB.Converter.output(item.authToken),
+							refreshToken: AWS.DynamoDB.Converter.output(item.authToken),
```
Comment (erodewald) [inline lib/DynamoDBContextStore.js]: This line appears to incorrectly set `refreshToken` to `authToken`.

File: README.md (hunk 1 above)
Comment (erodewald) [reply README.md]: Yes these should be removed from what I understand.

**Review by erodewald — CHANGES_REQUESTED** [review]
Personally very excited for this additional functionality. Storing arbitrary data in `state` is hugely useful to config.

File: README.md (hunk 1 above)
Comment (bflorian) [reply README.md]: :thumbsup:

File: lib/DynamoDBContextStore.js (hunk 2 above)
Comment (bflorian) [reply lib/DynamoDBContextStore.js]: :thumbsup:

**Comment by bflorian** [comment]
Made requested changes

**Review by erodewald — APPROVED** [review]

**Pull request merged**